File: proposed-projects/round-trip-experiment/reverse/20260909T125752Z_09b2c811/20260910T072624Z_d6bd66eb/ported_implementation/gbnf/utils/errors/get_input_as_string.py
"""Render any accepted input shape back to a string."""

from typing import List, Union

ValidInput = Union[str, int, List[int]]


def get_input_as_string(src: ValidInput) -> str:
    if isinstance(src, str):
        return src
    if isinstance(src, int):
        return chr(src)
    return "".join(chr(code_point) for code_point in src)

File: packages/gbnf-experiment/data/20260909T144331Z_5b46f630/ported_implementation/src/utils/errors/getInputAsString.ts
import type { ValidInput } from './errorsTypes.js';

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  if (typeof src === 'number') {
    return String.fromCodePoint(src);
  }
  return src.map((codePoint) => String.fromCodePoint(codePoint)).join('');
};
